mock/secrets_manager_client.go

```diff
@@ -3,46 +3,186 @@ package mock
+	DeleteSecretOutput *secretsmanager.DeleteSecretOutput
+}
 
 // CreateSecret saves the input options and returns a new mock secret. The mock
-// output can be customized. By default, it will create a cached mock secret
-// based on the input.
+// output can be customized. By default, it will create and save a cached mock
+// secret based on the input in the global secret cache.
 func (c *SecretsManagerClient) CreateSecret(ctx context.Context, in *secretsmanager.CreateSecretInput) (*secretsmanager.CreateSecretOutput, error) {
-	return nil, errors.New("TODO: implement")
+	c.CreateSecretInput = in
+
+	if c.CreateSecretOutput != nil {
+		return c.CreateSecretOutput, nil
+	}
+
+	if in.Name == nil {
+		return nil, errors.New("missing secret name")
+	}
+	if in.SecretBinary != nil && in.SecretString != nil {
+		return nil, errors.New("cannot specify both secret binary and secret string")
+	}
+	if in.SecretBinary == nil && in.SecretString == nil {
+		return nil, errors.New("must specify either secret binary or secret string")
+	}
+
+	if in.SecretBinary != nil {
+		GlobalSecretCache[*in.Name] = StoredSecret{
+			BinaryValue: in.SecretBinary,
+			Created:     time.Now(),
+		}
+	}
+	if in.SecretString != nil {
+		GlobalSecretCache[*in.Name] = StoredSecret{
+			Value:   *in.SecretString,
+			Created: time.Now(),
+		}
+	}
+
+	return &secretsmanager.CreateSecretOutput{
+		ARN:  in.Name,
+		Name: in.Name,
+	}, nil
 }
 
 // GetSecretValue saves the input options and returns an existing mock secret's
 // value. The mock output can be customized. By default, it will return a cached
-// mock secret if it exists.
+// mock secret if it exists in the global secret cache.
 func (c *SecretsManagerClient) GetSecretValue(ctx context.Context, in *secretsmanager.GetSecretValueInput) (*secretsmanager.GetSecretValueOutput, error) {
-	return nil, errors.New("TODO: implement")
+	c.GetSecretValueInput = in
+
+	if c.GetSecretValueOutput != nil {
+		return c.GetSecretValueOutput, nil
+	}
+
+	if in.SecretId == nil {
+		return nil, errors.New("missing secret ID")
+	}
+
+	s, ok := GlobalSecretCache[*in.SecretId]
+	if !ok {
+		return nil, errors.New("secret not found")
+	}
+
+	return &secretsmanager.GetSecretValueOutput{
+		Name:         in.SecretId,
+		ARN:          in.SecretId,
+		SecretString: aws.String(s.Value),
+		SecretBinary: s.BinaryValue,
+		CreatedDate:  aws.Time(s.Created),
+	}, nil
 }
 
 // UpdateSecretValue saves the input options and returns an updated mock secret
 // value. The mock output can be customized. By default, it will update a cached
-// mock secret if it exists.
+// mock secret if it exists in the global secret cache.
 func (c *SecretsManagerClient) UpdateSecretValue(ctx context.Context, in *secretsmanager.UpdateSecretInput) (*secretsmanager.UpdateSecretOutput, error) {
-	return nil, errors.New("TODO: implement")
+	c.UpdateSecretInput = in
+
+	if c.UpdateSecretOutput != nil {
+		return c.UpdateSecretOutput, nil
+	}
+
+	if in.SecretId == nil {
```

Comment:
Same as above, should this be a separate validate func?

Reply:
Same as above - probably not worth it unless validation gets more complicated.